### PR TITLE
Feature/aeternity to store (refactoring for the SSR)

### LINKED ISSUE #909: Don't import store explicitly
https://github.com/aeternity/superhero-ui/blob/78572cc2834215b08900226e92790cc230dffb6f/src/utils/aeternity.js#L11

https://github.com/aeternity/superhero-ui/blob/ab19bbfc47534148b432e252e13eb5f39b9e4ae7/src/utils/backend.js#L2

Probably the best would be to rewrite backend.js and aeternity.js as Vuex modules. This refactoring is needed for SSR.

### PR DESCRIPTION
**Pull request by shapkarin**
Refactoring the project. Move all possible Aeternity utils to the store by the Vuex module.

Hello @Liubov-crypto 
Please test:
- tippting (with extension and web wallet)
- retip (with extension and web wallet)
- comment (with extension and web wallet)
- it is shows right account balance (with extension)

--------------

close #909

### CODE REVIEW

File: src/store/index.js [hunk 1]
```diff
@@ -28,11 +29,6 @@ export default new Vuex.Store({
     tokenInfo: {},
     tokenBalances: [],
     isHiddenContent: true,
-    useSdkWallet: false,
-    useIframeWallet: false,
-    sdk: null,
-    isBackendLive: true,
-    cookiesConsent: { },
```
Comment (davidyuk) [inline src/store/index.js:35]: I think this is not heavily related to aeternity module, proposing to leave it here

**Review by davidyuk — CHANGES_REQUESTED** [review]
Not all references to `useIframeWallet`, `useSdkWallet` are fixed.
https://github.com/aeternity/superhero-ui/blob/d26a7ced134fda1b24d4c884d8cc319d61eb3a1c/src/components/layout/RightSectionWallet.vue#L99
https://github.com/aeternity/superhero-ui/blob/8a511ce5d4644b02ffa14ae19ac5d2fb37103d82/src/App.vue#L60

File: package-lock.json [hunk 2]
```diff
@@ -972,6 +972,7 @@
       "version": "7.8.4",
       "resolved": "https://registry.npmjs.org/@babel/runtime/-/runtime-7.8.4.tgz",
       "integrity": "sha512-neAp3zt80trRVBI1x0azq6c57aNBqYZH8KhMm3TaB7wEI5Q4A2SHfBHE8w9gOhI/lrqxtEbXZgQIrHP+wvSGwQ==",
+      "dev": true,
```
[the remark this answers is not in the document]
Comment (shapkarin) [reply package-lock.json]: fixed

**Comment by shapkarin** [comment]
@Liubov-crypto please, check it now: Deployed to feature-aeternity-to-store.aggregator.z52da5wt.xyz

File: src/store/modules/aeternity.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+      if (!sdk) throw new Error('Init sdk first');
+      if (isTippingContractsInitialised) return;
+      const [contractV1, contractV2, contractV3] = await Promise.all([
+        [TIPPING_INTERFACE_V1, process.env.VUE_APP_CONTRACT_V1_ADDRESS],
+        [TIPPING_INTERFACE_V2, process.env.VUE_APP_CONTRACT_V2_ADDRESS],
+        [TIPPING_INTERFACE_V3, process.env.VUE_APP_CONTRACT_V3_ADDRESS],
+      ].map(([tippingInterface, contractAddress]) => (contractAddress
+        ? sdk.getContractInstance(tippingInterface, { contractAddress })
+        : null)));
+      commit('isTippingContractsInitialised', true);
```
Comment (davidyuk) [inline src/store/modules/aeternity.js:62]: I can't find where this mutation is defined

File: src/store/modules/aeternity.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+      commit('setContractV1', contractV1);
+      commit('setContractV2', contractV2);
+      commit('setContractV3', contractV3);
```
Comment (davidyuk) [inline src/store/modules/aeternity.js:65]: Would be nice to combine this into a single mutation, but it is not necessary to fix it now

File: src/store/modules/aeternity.js [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+            accounts: [
+              MemoryAccount({
+                keypair: { secretKey: Cypress.env('privateKey'), publicKey: Cypress.env('publicKey') },
+              }),
+            ],
+            address: Cypress.env('publicKey'),
+          });
+          const rpcClient = async () => Cypress.env('publicKey');
+
+          commit('setSDK', { instance, rpcClient });
```
Comment (davidyuk) [inline src/store/modules/aeternity.js:85]:
```suggestion
          instance.rpcClient = { getCurrentAccount: async () => Cypress.env('publicKey') };

          commit('setSDK', instance);
```
Not needed to inject rpcClient inside setSDK

File: src/store/modules/aeternity.js [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+        } else {
+          const instance = await RpcAepp({
+            ...options,
+            name: 'Superhero',
+            onDisconnect() {
+              commit('resetState');
+            },
+          });
+          commit('setSDK', { instance });
+          await dispatch('initTippingContractIfNeeded');
```
Comment (davidyuk) [inline src/store/modules/aeternity.js:96]: Didn't get why do you need it, why it was working before without it?

File: src/store/modules/aeternity.js [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+        } else {
+          const instance = await RpcAepp({
+            ...options,
+            name: 'Superhero',
+            onDisconnect() {
+              commit('resetState');
+            },
+          });
+          commit('setSDK', { instance });
```
Comment (davidyuk) [inline src/store/modules/aeternity.js:95]: Can you call setSdk one at the end as it was before?

**Review by davidyuk — COMMENTED** [review]
> Not all references to `useIframeWallet`, `useSdkWallet` are fixed

As I see, this is still not fixed

File: src/store/modules/aeternity.js (hunk 3 above)
Comment (shapkarin) [reply src/store/modules/aeternity.js:62]: thanks!

File: src/store/modules/aeternity.js [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+    setContractV1(state, contract) {
+      state.contractV1 = contract;
+    },
+    setContractV2(state, contract) {
+      state.contractV2 = contract;
+    },
+    setContractV3(state, contract) {
+      state.contractV3 = contract;
+    },
+    setSDK(state, { instance, rpcClient: getCurrentAccount }) {
```
[the remark this answers is not in the document]
Comment (shapkarin) [reply src/store/modules/aeternity.js:35]: Didn't thought about such problem. Thanks. I will revert it.

File: src/store/modules/aeternity.js (hunk 4 above)
Comment (shapkarin) [reply src/store/modules/aeternity.js:65]: I did that at the first time, but thought that it is overhead. No problem, will commit soon.

Comment (shapkarin) [reply src/store/modules/aeternity.js:65]: done

File: src/store/modules/aeternity.js (hunk 5 above)
Comment (shapkarin) [reply src/store/modules/aeternity.js:85]: Make sense.

File: src/store/modules/aeternity.js (hunk 6 above)
Comment (shapkarin) [reply src/store/modules/aeternity.js:96]: to be honest, I'm still not sure. Looks like that's depends on some other init nature after the refactoring. Without this contract is null...

File: src/store/modules/aeternity.js (hunk 7 above)
Comment (shapkarin) [reply src/store/modules/aeternity.js:95]: if I need to use `initTippingContractIfNeeded` at the same function, so I need to call in before the  
```js
await dispatch('initTippingContractIfNeeded');
```

File: src/store/modules/aeternity.js [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+    sdk: null,
+    useSdkWallet: false,
+    useIframeWallet: false,
+    contractV1: null,
+    contractV2: null,
+    contractV3: null,
+    isTippingContractsInitialised: false,
+  },
+  mutations: {
+    setAllContracts(state, contracts) {
```
Comment (davidyuk) [inline src/store/modules/aeternity.js:26]:
```suggestion
    setContracts(state, contracts) {
```
🤷‍♀️

File: src/store/modules/aeternity.js [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+      const [contractV1, contractV2, contractV3] = contracts;
+      state.contractV1 = contractV1;
+      state.contractV2 = contractV2;
+      state.contractV3 = contractV3;
```
Comment (davidyuk) [inline src/store/modules/aeternity.js:30]:
```suggestion
      state.contractV3 = contractV3;
      state.isTippingContractsInitialised = true;
```
`isTippingContractsInitialised` is not changed separately, so I think we can do it at the same time.

**Review by davidyuk — APPROVED** [review]
Code changes look acceptable, I think it is ready to be tested, cc: @Liubov-crypto

File: src/store/modules/aeternity.js (hunk 10 above)
Comment (shapkarin) [reply src/store/modules/aeternity.js:30]: OK, so I don't think we should keep the setTippingContractsInitialised mutation too.

File: src/store/modules/aeternity.js (hunk 9 above)
Comment (shapkarin) [reply src/store/modules/aeternity.js:26]: that's better

**Comment by Liubov-crypto** [comment]
retip (with extension) - pass
comment (with extension) - pass
it is shown right account balance (with extension) - pass
Unable to check other transactions because the URL has a connection issue (maintenance)

**Comment by Liubov-crypto** [comment]
tipping (with web wallet) - pass
retip (with web wallet) - pass
comment (with web wallet) - pass

Login with Web Wallet button is not active on https://feature-aeternity-to-store.aggregator.z52da5wt.xyz/ 
I can't log in to iframe wallet with this button: 
button not active

**Comment by shapkarin** [comment]
@Liubov-crypto looks like you need to wait for a some time or use testnet also in еру wallet's settings.

**Comment by shapkarin** [comment]
@davidyuk can you please have a look. for me it looks OK now, please check.

**Comment by shapkarin** [comment]
@davidyuk I see.
> davidyuk approved these changes (17 hours ago)

I rebased it

**Comment by shapkarin** [comment]
@davidyuk please notice that current branch uses testnet for the tests before the merge.
@Liubov-crypto please, recheck, I guess Denis explained what the problems it was yesterday and they are not related to this refactoring.

**Comment by Liubov-crypto** [comment]
tipping (with web wallet) - pass
retip (with web wallet) - pass
comment (with web wallet) - pass

This issue is still reproducible.
Login with Web Wallet button is not active on https://feature-aeternity-to-store.aggregator.z52da5wt.xyz/
I can't log in to iframe wallet with this button:
button not active

tipping (with extension) - failed
retip (with extension) - failed
comment (with extension) - pass
it is shown right account balance (with extension) - pass (only on the testnet) and incorrect balance on the mainnet. 
retip error 1
errors in feed

Also connectivity issues:


I'll check again when the connection is stable.

**Pull request merged**